docs/classes/request.md

```diff
@@ -2,86 +2,128 @@
 
 * Table of Contents
 {:toc}
 
-The `Request` interface generally finds its only use in responder callables (or [`Websocket::onOpen()`](websocket.html#onopenint-clientid-handshakedata)). [`Middleware`s](middleware.md) do never see the `Request`; the `StandardRequest` class is supposed to be a simple request API reading from and manipulating [`InternalRequest`](internalrequest.md) under the hood.
+## `Constructor`
+
+```php
+public function __construct(
+    Client $client, 
+    string $method, 
+    Psr\Http\Message\UriInterface $uri, 
+    string[] | string[][] $headers = [], 
+    RequestBody | Amp\ByteStream\InputStream | string | null $body = null, 
+    string $protocol = "1.1"
+)
+```
+
+### Parameters
+
+|[`Client`](client.md)|`$client`|The client sending the request|
+|`string`|`$method`|HTTP request method|
+|[`Psr\Http\Message\UriInterface`](https://www.php-fig.org/psr/psr-7/#35-psrhttpmessageuriinterface)|`$uri`|The full URI being requested, including host, port, and protocol|
+|`string[]`<br />`string[][]`|`$headers`|An array of strings or an array of string arrays.|
+|[`RequestBody`](requestbody.md)<br />[`Amp\ByteStream\InputStream`](https://amphp.org/byte-stream/)<br />`string`<br />`null`|`$body`|Request body|
+|`string`|`$protocol`|HTTP protocol version|
+
+## `getClient(): Client`
+
+Returns the [`client`](client.md) sending the request
 
 ## `getMethod(): string`
 
-Returns the used method, e.g. `"GET"`.
+Returns the HTTP method used to make this request, e.g. `"GET"`.
+
+## `setMethod(string $method)`
+
+Sets the request HTTP method.
+
+## `getUri(): Psr\Http\Message\UriInterface`
 
-## `getUri(): string`
+Returns the request [`URI`](https://www.php-fig.org/psr/psr-7/#35-psrhttpmessageuriinterface).
 
-Returns the requested URI (path and query string), e.g. `"/foo?bar"`.
+## `setUri(Psr\Http\Message\UriInterface $uri)`
+
+Sets a new [`URI`](https://www.php-fig.org/psr/psr-7/#35-psrhttpmessageuriinterface) for the request.
 
 ## `getProtocolVersion(): string`
 
-Currently it will return one of the three supported versions: `"1.0"`, `"1.1"` or `"2.0"`.
+Returns the HTTP protocol version as a string (e.g. "1.0", "1.1", "2.0").
 
-## `getHeader(string): string | null`
+## `setProtocolVersion(string $protocol)`
 
-Gets the first value of all the headers with that name.
+Sets a new protocol version number for the request.
 
-## `getHeaderArray(string): array<string>`
+## `setHeader(string $name, string | string[] $value)`
 
-Gets an array with headers. HTTP allows for multiple headers with the same name, so this returns an array. Usually only a single header is needed and expected, in this case there is `getHeader()`.
+Sets the named header to the given value.
 
-## `getAllHeaders(): array<array<string>>`
+If header `$name` is not valid or `$value` is invalid, [`\Error`](http://php.net/manual/en/class.error.php) is thrown. 
```

Comment:
This only happens when assertions are enabled, currently.

Reply:
Should I leave info block about it (with details about `assert.exception` settings) or it's better to remove it at all?

Reply:
Not sure, generally `Error` instances shouldn't be caught, anyway. @trowski Thoughts?

Reply:
I'd just leave it out of the docs for now as @kelunik said `Error` shouldn't be caught anyway. Both incoming and outgoing header blocks are validated regardless of the assertion setting, we only have the assertions there so the error is thrown in a relevant location during development.